caluma/caluma_core/validations.py

```diff
@@ -48,15 +45,20 @@ class BaseValidation(object):
     """
 
     def __init__(self):
-        validation_fns = inspect.getmembers(self, lambda m: hasattr(m, "_validation"))
-        validation_muts = [fn._validation.__name__ for _, fn in validation_fns]
+        validation_fns = inspect.getmembers(self, lambda m: hasattr(m, "_validations"))
+
+        validation_muts = [
+            val.__name__ for _, fn in validation_fns for val in fn._validations
+        ]
         validation_muts_dups = list_duplicates(validation_muts)
         if validation_muts_dups:
             raise ImproperlyConfigured(
                 f"`validation_for` defined multiple times for "
                 f"{', '.join(validation_muts_dups)} in {str(self)}"
             )
-        self._validations = {fn._validation: fn for _, fn in validation_fns}
+        self._validations = {
+            validation: fn for _, fn in validation_fns for validation in fn._validations
```

Comment:
I don't see the problem here, there is only one function allowed per validation. This is checked above and throws an `ImproperlyConfigured` exception otherwise

Reply:
Right! It was the naming that threw me off here. 

What I'd suggest is renaming the iterator variable in the comprehension above to avoid confusion:
```python
        validation_muts = [
            mut.__name__ for _, fn in validation_fns for mut in fn._validations
        ] 
```
Same of course for the visibilities

Reply:
Can do :)